### DISCUSSION TITLE
Let's discuss having saner default connection settings for Transmission

### DISCUSSION THREAD

**Discussion by Pentaphon** [transmission/transmission#3460] [Ideas]
I think we as a community should come up with better default connection settings for Transmission, as we are about to get an influx of new users when 4.0 becomes stable. 

- **peer-limit-per-torrent: Number (default = 60)** This is probably enough for public torrents, which most people use. Others might disagree. I've seen people set this to 100 most of the time. 
- **peer-limit-global: Number (default = 240)** This should probably be raised to something bigger since more people have better routers and are seeding more torrents at once. Perhaps 500?
- **upload-slots-per-torrent: Number (default = 14)** I think this is too high and just leads to people's upload speed on a single torrent (which is usually set too low by many) to be divided by 14 instead of something reasonable like 4, which leads to Transmission users not uploading quickly to their connected leechers. Imagine seeding at say 500 kB/s and your torrent is going to 14 users at 35 kB/s instead of 4 users at 125 kB/s. This makes a massive difference to a big swarm as users are more likely to get more data from Transmission users. I've made this an issue here: https://github.com/transmission/transmission/issues/3459

Thoughts?

**Comment by GaryElshaw** [comment]
I think Charles has something similar in mind with #3372.

**Reply by ckerr** [reply]
No, 3372 is a different thing entirely -- It's about allocating how many requests we send to each peer.

**Comment by ckerr** [comment]
IMO the 60 / 240 numbers still seem reasonable in 2022 IMO. I doubt that casual users are ever going to hit those limits, so raising them seems moot.

I agree 14 might be a little high, for the reasons you cite about swarm health.

**Reply by Pentaphon** [reply]
> IMO the 60 / 240 numbers still seem reasonable in 2022 IMO. I doubt that casual users are ever going to hit those limits, so raising them seems moot.

Yes, I'm not even worried about those values because at least people can easily change them in the GUI.

> I agree 14 might be a little high, for the reasons you cite about swarm health.

Thanks! I think this is the biggest issue right now because you can't change it in the GUI and editing a settings.json file is way too "scary" for most people to do, so changing the default value to something more swarm-friendly is essential to the overall torrent community by making Transmission a more efficient seeding client.

**Comment by Pentaphon** [comment]
Looks like this discussion bore fruit thanks to Gary whose PR: https://github.com/transmission/transmission/pull/3474 is fixing the default number of slots per torrent. Thanks to Charles for being open to this seldom-discussed yet important issue being fixed in time  for 4.0.0.

**Reply by GaryElshaw** [reply]
Happy to help.
